src/App.tsx

```diff
@@ -1,73 +1,99 @@
+    applyQuery(event.target.value);
+  };
+
+  useEffect(() => {
+    if (inputValue.current) {
+      inputValue.current.focus();
+    }
+  }, []);
+
+  const filteredPeople = useMemo(() => {
+    if (appliedQuery === '') {
+      return people;
+    }
+
+    return people.filter(person =>
+      person.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+    );
+  }, [people, appliedQuery]);
+
+  const handleChooseUser = useCallback((person: Person) => {
+    setSelectedPerson(person);
+  }, []);
+
+  const visiblePeople = filteredPeople;
 
   return (
     <div className="container">
       <main className="section is-flex is-flex-direction-column">
-        <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
-        </h1>
+        {selectedPerson ? (
+          <h1 className="title" data-cy="title">
+            {`${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`}
+          </h1>
+        ) : (
+          <h1 className="title" data-cy="title">
+            No selected person
+          </h1>
+        )}
 
         <div className="dropdown is-active">
           <div className="dropdown-trigger">
             <input
               type="text"
+              value={query}
+              ref={inputValue}
+              onChange={handleQueryChange}
               placeholder="Enter a part of the name"
               className="input"
               data-cy="search-input"
             />
           </div>
 
-          <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
-            <div className="dropdown-content">
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Bernard Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Antone Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter de Decker</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Petronella de Decker</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Hercke</p>
-              </div>
-            </div>
-          </div>
+          <DropDown
+            visiblePeople={visiblePeople}
+            appliedQuery={appliedQuery}
+            onSelected={handleChooseUser}
+          />
```

Comment:
Ensure that the `DropDown` component correctly handles the `visiblePeople`, `appliedQuery`, and `onSelected` props. Verify that the `DropDown` component's implementation matches the expected usage here to prevent any runtime errors.